### DISCUSSION TITLE
Sea ORM Mismatch Custom JSON Type

### DISCUSSION THREAD

**Discussion by niitou** [SeaQL/sea-orm#1419] [Q&A]
Hi, i'm running into an issue where my entity model producing an error when fetched using a query. Below is the error message.

`DbErr: Query Error: error occurred while decoding column "animeseason": mismatched types; Rust type 'core::option::Option<sqlx_core::types::json::Json<entity::manami_anime::AnimeSeason>>' (as SQL type 'JSONB') is not compatible with SQL type 'animeSeason'`

And below is my entity model

```
#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, FromJsonQueryResult)]
pub struct AnimeSeason {
    pub season: Season, // Enum
    pub year : i64
}

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "manami-anime", schema_name="public")]
pub struct Model{
    #[sea_orm(primary_key)]
    pub id : i32,
    pub sources : Vec<String>,
    pub title : String,
    #[sea_orm(nullable)]
    pub episodes : i64,
    pub picture : String,
    pub thumbnail : String,
    #[sea_orm(nullable)]
    pub synonyms : Vec<String>,
    #[sea_orm(nullable)]
    pub relations : Vec<String>,
    pub tags : Vec<String>,
    #[sea_orm(default_value="UNKNOWN")]
    pub status : Status,
    pub animeseason : AnimeSeason,
    #[sea_orm(default_value="UNDEFINED")]
    pub animetype : Type
}
```

This is the type on my postgres database.
![image](https://user-images.githubusercontent.com/57691998/213992531-c57932ab-0519-465f-8e9d-8049f28a372d.png)

**Comment by billy1624** [comment]
Hey @StevenBudi, can I take a look at the source code if it's possible. So I can test it and reproduce the example.
